workflow-trace-viewer/src/jvmMain/kotlin/com/squareup/workflow1/traceviewer/LandingWindow.kt

```diff
@@ -0,0 +1,124 @@
+package com.squareup.workflow1.traceviewer
+
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.padding
+import androidx.compose.material.MaterialTheme
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.produceState
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import com.squareup.workflow1.traceviewer.ui.control.DisplayDevices
+import com.squareup.workflow1.traceviewer.ui.control.UploadFile
+import io.github.vinceglb.filekit.PlatformFile
+import kotlinx.coroutines.delay
+
+/**
+ * Main window composable that shows both file upload and device selection options.
+ */
+@Composable
+internal fun LandingWindow(
+  modifier: Modifier = Modifier,
+  onFileSelected: (PlatformFile) -> Unit,
+  onDeviceSelected: (String) -> Unit
+) {
+  val devices by produceState(initialValue = listDevices()) {
+    while (true) {
+      delay(3000L)
+      value = listDevices()
+    }
```

Comment:
😱 . what is this about? we have to wait to get an adb connection or something?

Reply:
If so, woudl be good to comment this delay and also use a constant on that magic `3000L`

Reply:
Previously the adb list was fetched when toggling live mode on, now we show the list of devices while continuously refreshing.